sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -12,19 +12,157 @@
 #include <string>
 
 namespace Azure { namespace Core {
+
+  /**
+   * @brief An exception that gets thrown when @DateTime error occurs.
+   */
+  class DateTimeException : public std::runtime_error {
+  public:
+    /**
+     * @brief Construct with message string.
+     *
+     * @param msg Message string.
+     */
+    explicit DateTimeException(std::string const& msg) : std::runtime_error(msg) {}
+  };
+
   /**
    * @brief Manages date and time in standardized string formats.
    */
   class DateTime {
   public:
-    /// A type that represents tick spans.
-    typedef uint64_t IntervalType;
-
-  private:
-    //  Number of seconds between 01-01-1970 and 01-01-1601.
-    static constexpr IntervalType WindowsToPosixOffsetSeconds = 11644473600LL;
+    /**
+     * @brief Represents time duration.
+     */
+    class Duration {
+      friend class DateTime;
+
+      int64_t m_100nsIntervals;
+      constexpr explicit Duration(int64_t intervalsOf100ns) : m_100nsIntervals(intervalsOf100ns) {}
+
+      static constexpr int64_t IntervalsOf100nsPerMicrosecond = 10;
+      static constexpr int64_t IntervalsOf100nsPerMillisecond
+          = 1000 * IntervalsOf100nsPerMicrosecond;
+      static constexpr int64_t IntervalsOf100nsPerSecond = 1000 * IntervalsOf100nsPerMillisecond;
+      static constexpr int64_t IntervalsOf100nsPerMinute = 60 * IntervalsOf100nsPerSecond;
+      static constexpr int64_t IntervalsOf100nsPerHour = 60 * IntervalsOf100nsPerMinute;
+
+    public:
+      constexpr Duration() : m_100nsIntervals(0) {}
+
+      static constexpr Duration FromHours(int hours)
+      {
+        return Duration(hours * IntervalsOf100nsPerHour);
+      }
+
+      void constexpr AddHours(int hours) { m_100nsIntervals += (hours * IntervalsOf100nsPerHour); }
+
+      static constexpr Duration FromMinutes(long long minutes)
```

Comment:
Why `long long` and not int, or just `int64_t`?

Reply:
Should we use explicit inttypes (http://www.cplusplus.com/reference/cstdint/) like int32_t and int64_t in public surface area instead of int/long long? We use the inttypes in C as well.

Reply:
See https://azure.github.io/azure-sdk/cpp_design.html#integer-sizes

Reply:
As I said, for the data that does not gets sent over the wire, I don't think we have to. But to move on for now, I updated the types.